Utest/rimea_tests/reportscript.py

```diff
@@ -289,10 +319,49 @@ def get_log(testnumber):
 		log_name = "./test_" + str(testnumber) + "/log_test_" + str(testnumber) + ".txt"
 		with open(log_name) as log:
 				return log.read().split("\n")[-3:-1]
+def get_diagrams(testnumber):
+	start_direc = os.getcwd()
+	direc = start_direc + '/test_' + str(testnumber)
+	os.chdir(direc)
+	data_list = os.listdir(direc)
+	i=0
+	arc=False
+	while i <= (len(data_list)-1) and arc!= True:
+		j = data_list[i].split('.')
+		if len(j) == 2:
+			if j[1] == 'png':
+				d_path = direc + '/' + str(j[0]) + '.png'
+				arc=True
+				title=j[0]
+			else:
+				arc=False
+		else:
+			arc=False
+		i=i+1
+	if arc == False:
+		d_path='_'
+		title='_'
+	os.chdir(start_direc)
+	return(d_path,arc,title)
```

Comment:
This function looks not like it is doing what it is supposed to be doing based on its name.
From the name I infer that it should return a collection of paths to all diagrams generated for a test based on the tests number. Looking at the code however it does something else:
My understanding is that the functions returns a tuple of: The absolute path of the first PNG file, a bool indicating if any PNG file was found, and a tile string (the file stem) or an underscore if no file was found.

Now I think this can be rewritten very nicely in s few steps. First a small cleanup w.r.t to path handling:
```python
start_direc = os.getcwd()
direc = start_direc + '/test_' + str(testnumber)
os.chdir(direc)
data_list = os.listdir(direc)
```
becomes
```python
test_directory = os.path.join(os.getcwd(), "test_{}".format(testnumer))
```
Note: no change dir

the while loop can now be rewritten as:
```python
diagrams=[]
for filename in os.listdir(directory):
    if filename.endswith(".png"): 
         diagrams.append((os.path.splitext(filename)[0], filename ))
return diagrams
```
or as list expression
```python
return [(os.path.splitext(f)[0], f ) for f in os.listdir(directory) if f.endswith(".png")]
```